### PR TITLE
Convert all files in this repository to LF (unix) line endings.

### PR DESCRIPTION
**Pull request by paulbartell**
Description
-----------
Convert all files in this repository to LF (unix) line endings.
Set text=auto for all files to ensure that files have correct line endings upon checkin.

Refer to the [github docs on line-endings](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings) for more information.

In general, git handles LF line endings in a much more sane way than it does CRLF line endings. Users of windows should see little change to the way the use this repository. By default, git with automatically convert to/from native line endings during checkout and commit operations.

For more info on these settings in git, check out the [git documentation](https://git-scm.com/docs/gitattributes#_text) on the subject.

Test Steps
-----------
Validated that new clones of the repository handle automatic line ending conversion correctly on windows and *nix.

Related Issue
-----------
#234 #340 #339 #247

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by cobusve — DISMISSED** [review]
Please make sure we test this with all the toolchains, last time we eliminated line endings we ran into some compilation errors.

**Comment by RichardBarry** [comment]
Why is this necessary?

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Fix trailing whitespace and newline at EOF
+483f9a08ca2a05cb60a0e838fec03b465cc56c26
```
Comment (archigup) [inline .git-blame-ignore-revs:2]: Won't these commit ids not work after squash?

Comment (paulbartell) [reply .git-blame-ignore-revs:2]: Not planning to squash this PR on purpose.

**Review by archigup — APPROVED** [review]

**Review by archigup — APPROVED** [review]

**Comment by kstribrnAmzn** [comment]
Did you do any other manual changesthan changing some .git files, a few readmes, and the CI to now check for the correct line ending? If the rest was automatically done, I'm happy to approve.

**Comment by paulbartell** [comment]
> Did you do any other manual changes than changing some .git files, a few readmes, and the CI to now check for the correct line ending? If the rest was automatically done, I'm happy to approve.

All automatic. Everything is in CI except for rejecting code with tabs. I can add that in at a later time.

**Review by kstribrnAmzn — APPROVED** [review]

**Review by kstribrnAmzn — APPROVED** [review]

**Review by archigup — APPROVED** [review]

**Pull request merged**